src/parser/Parser.ts

```diff
@@ -2449,6 +2550,15 @@ export class Parser {
                 if (!isClassMethodStatement(parent)) {
                     this._references.functionExpressions.push(expression);
                 }
+                currentFunctionExpression = expression;
```

Comment:
With full recursive walk this is going to be wrong with nested functions:
- enter first function, `currentFunctionExpression=A`
- new var belongs to `A`
- enter sub function, `currentFunctionExpression=B`
- new var belongs to `B`
- exit sub function, `currentFunctionExpression=B`
- new var belongs to `B` though we're in `A`

Reply:
We'd need to change the walking to be non-recursive, of do this walking "late" using a non-recursive walk just when needed.

Reply:
Ah, great catch! That's what I get for skipping writing unit tests for this! 

I refactored and added some tests in [this commit](https://github.com/rokucommunity/brighterscript/pull/258/commits/73c432c8485f870ddb0c10d36765269ca302d0b2). Is that better?

Reply:
Looks sensible now. Good idea to use the "growing" list of function expressions as they are discovered.